plugins/web/opentelemetry-instrumentation-web-exception/src/instrumentation.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+import {
+  ATTR_EXCEPTION_TYPE,
+  ATTR_EXCEPTION_MESSAGE,
+  ATTR_EXCEPTION_STACKTRACE,
+} from '@opentelemetry/semantic-conventions';
+// TODO: should events be imported or included as part of the instrumentation base class?
+import { events } from '@opentelemetry/api-events';
+import { SeverityNumber } from '@opentelemetry/api-logs';
+import { Attributes } from '@opentelemetry/api';
+
+export interface GlobalErrorsInstrumentationConfig
+  extends InstrumentationConfig {
+  /**
+   * A callback function for adding custom attributes to the span when an error is recorded.
+   *
+   * @param {Error} error - The error object that is being recorded.
+   * @returns {Attributes} - The attributes to add to the span.
+   */
+  applyCustomAttributes?: (error: Error) => Attributes;
+}
+
+export class WebExceptionInstrumentation extends InstrumentationBase<GlobalErrorsInstrumentationConfig> {
+  readonly applyCustomAttributes?: (error: Error) => Attributes;
+  constructor(config: GlobalErrorsInstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-web-exception', '0.0.1', config);
+
+    this.applyCustomAttributes = config.applyCustomAttributes;
+  }
+
+  init() {}
+
+  onError = (event: ErrorEvent | PromiseRejectionEvent) => {
+    const error: Error | undefined =
+      'reason' in event ? event.reason : event.error;
+    if (error) {
+      const message = error.message;
+      const type = error.name;
+      const errorAttributes = {
+        [ATTR_EXCEPTION_TYPE]: type,
+        [ATTR_EXCEPTION_MESSAGE]: message,
+        [ATTR_EXCEPTION_STACKTRACE]: error.stack,
+      };
+
+      const eventLogger = events.getEventLogger(
+        this.instrumentationName,
+        this.instrumentationVersion
+      );
+
+      const customAttributes = this.applyCustomAttributes
+        ? this.applyCustomAttributes(error)
+        : {};
+
+      eventLogger.emit({
+        name: 'exception',
```

Comment:
Event names should be namespaced so just `exception` doesn't actually match the semantic conventions.

Please work with @Karlie-777 and her PR around defining this exception https://github.com/open-telemetry/semantic-conventions/pull/1942. I agree that using the 3 top level attributes makes sense for these and I've left some comments in the link PR as well.

There are a few other properties that we should also be setting (if possible), which does mean attempting to parse out the exception on the client.

Reply:
Could we be consistent with the [existing semantic conventions for span events](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md#exception-event) (where the event name is "exception")? Or, do we need to define exception event for each separate domain (e.g. "browser.exception", "android.exception" etc.)?

Reply:
I've changed it to `exception` for now since we're not creating any client specific semantic conventions for errors yet. We can change it in the future if this pattern develops.